lib/trip_dispatcher.rb

```diff
@@ -90,6 +89,87 @@ def load_trips
       trips
     end
 
+    def find_available_drivers
+      available_drivers = @drivers.find_all do |driver|
+        driver.status == :AVAILABLE
```

Comment:
I love that you've broken this  and `select_driver` out as a separate helper methods! This makes the code in this file much easier to read.